node_modules/oae-content/lib/internal/util.js

```diff
@@ -104,10 +106,35 @@ var augmentContent = module.exports.augmentContent = function(ctx, content, dura
             content.previews.wideUrl = getSignedDownloadUrl(ctx, content.previews.wideUri, duration, offset);
             delete content.previews.wideUri;
         }
+
+        if (content.previews.video) {
+            content.previews.video = _getSignedVideoUrl(ctx, content.previews.key);
+        }
     }
 };
 
 /**
+ * Generate a signed url that can be used to show video previews in the browser for a video file stored in AWS S3
+ *
+ * @param  {Context}    ctx         Standard context object containing the current user and the current tenant
+ * @param  {String}     key         The name of the file stored in S3
+ * @return {String}                 The signed S3 url that can be used to view the video
+ */
+var _getSignedVideoUrl = function(ctx, key) {
+    var config = ZencoderUtil.getConfig(ctx.tenant().alias);
+
+    AWS.config.update({
+        'accessKeyId': config['amazons3-access-key'],
+        'secretAccessKey': config['amazons3-secret-key'],
+        'region': config['amazons3-region']
+    });
+
+    var s3 = new AWS.S3();
+
+    return s3.getSignedUrl('getObject', { Bucket: config['amazons3-bucket'], Key: key, Expiration: TIME_1_WEEK_IN_SECONDS });
```

Comment:
Also, if this involves a remote network call, it should use an asynchronous call rather than synchronous.